### ISSUE TITLE
Merge in a more restrictive constraint

### ISSUE THREAD

**Issue by himito** [uqfoundation/mystic#177]
Hi,

Firstly, thank you for developing this useful library :)
I would like to know if it's possible to compute a more restrictive constraint from a set of constraints. I mean

```
> f(*['x0 > 5', 'x0 > 7', 'y0 < 4', 'y0 == 1'])
('x0 > 7', 'y0 == 1')
```

Thank you in advance.

Cheers,
Jaime

**Comment by mmckerns** [comment]
Yeah, `merge` is supposed to do something like this... but is currently limited to exact matches.
```
>>> import mystic as my
>>> my.symbolic.merge(*['x < 7', 'x > 7'])
('x != 7',)
>>> my.symbolic.merge(*['x > 5', 'x > 7'])  # currently doesn't reduce
('x > 5', 'x > 7')
```
It feels like `merge` should reduce the latter case as well. So, I'll take your question as a feature request.

**Comment by himito** [comment]
Thank you @mmckerns

**Comment by mmckerns** [comment]
This applies to `condense` as well... as noted in the current source code:
```
#FIXME: should be better, currently only condenses exact matches
def condense(*equations, **kwds):
```

**Comment by mmckerns** [comment]
It looks like `sympy` has improved their inequality solvers since I implemented much of the symbolic module... so maybe I can leverage sympy.solvers.inequalities.reduce_inequalities and other recent tools in https://docs.sympy.org/latest/modules/solvers/inequalities.html.
